src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -2253,6 +2311,20 @@ def set_up_node_restriction(self, mc: ManagedCluster) -> ManagedCluster:
 
         return mc
 
+    def set_up_vpa(self, mc: ManagedCluster) -> ManagedCluster:
+        """Set up workload auto-scaler profile vpa for the ManagedCluster object.
+
+        :return: the ManagedCluster object
+        """
+        self._ensure_mc(mc)
+
+        if self.context.get_enable_vpa():
+            if mc.workload_auto_scaler_profile is None:
+                mc.workload_auto_scaler_profile = self.models.ManagedClusterWorkloadAutoScalerProfile()
+            mc.workload_auto_scaler_profile.vertical_pod_autoscaler = self.models.ManagedClusterWorkloadAutoScalerProfileVerticalPodAutoscaler(enable=True)
```

Comment:
Check if mc.workload_auto_scaler_profile.vertical_pod_autoscaler is None?

Reply:
Updated in the latest iteration